vertx-mysql-client/src/main/java/io/vertx/mysqlclient/impl/codec/MySQLDecoder.java

```diff
@@ -14,68 +14,65 @@
 import io.netty.buffer.ByteBuf;
 import io.netty.buffer.CompositeByteBuf;
 import io.netty.channel.ChannelHandlerContext;
-import io.netty.handler.codec.ByteToMessageDecoder;
-import io.netty.util.ReferenceCountUtil;
-import io.vertx.mysqlclient.impl.MySQLSocketConnection;
+import io.netty.channel.ChannelInboundHandlerAdapter;
 
 import java.util.ArrayDeque;
-import java.util.List;
 
-import static io.vertx.mysqlclient.impl.protocol.Packets.*;
+import static io.vertx.mysqlclient.impl.protocol.Packets.PACKET_PAYLOAD_LENGTH_LIMIT;
 
-class MySQLDecoder extends ByteToMessageDecoder {
+class MySQLDecoder extends ChannelInboundHandlerAdapter {
```

Comment:
you should look at the PgDecoder that is very similar, it is written in a way that it will slice packet itself without using `LengthFieldBasedFrameDecoder` which saves an inbound handler in the channel pipeline.

Reply:
Let's address that in a separate PR. I got rid of the last `slice()` calls in this decoder in 36574ee1.